aodntools/timeseries_products/hourly_timeseries.py

```diff
@@ -0,0 +1,581 @@
+from __future__ import print_function
+import sys
+import os.path
+from collections import OrderedDict
+from dateutil.parser import parse
+from datetime import datetime
+import json
+from netCDF4 import Dataset
+import argparse
+
+import numpy as np
+import xarray as xr
+import pandas as pd
+
+
+def sort_files_to_aggregate(files_to_aggregate):
+    """
+    sort the list of files to aggregate by time_deployment start attribute
+
+    :param files_to_aggregate: list of file URLs
+    :return: list of file URLs
+    """
+    file_list_dataframe = pd.DataFrame(columns=["url", "deployment_date"])
+    for file in files_to_aggregate:
+
+        with Dataset(file) as nc:
+            global_attributes = nc.ncattrs()
+            if 'time_deployment_start' in global_attributes:
+                deployment_date = nc.getncattr('time_deployment_start')
+                file_list_dataframe = file_list_dataframe.append({'url': file,
+                                                                  'deployment_date': parse(deployment_date)},
+                                                                 ignore_index=True)
+            else:
+                print('NO time_deployment_start attribute in file ' + file)
```

Comment:
As I said, we should be doing this check in `check_file()`, and dealing with these missing attributes the same way as other issues that prevent files from being included in the product.

Reply:
I've made a new file checker function that do the checks and return a chronologically sorted list of file urls and a dict of files and its errors. Tha replaces the old file checker and the files sorter function.